backend/dataall/modules/redshift_datasets_shares/services/redshift_table_share_processor.py

```diff
@@ -603,3 +604,147 @@ def verify_shares(self) -> bool:
                     )
         return True
+
+    def cleanup_shares(self) -> bool:
+        """
+        For each table:
+            Update table status with Start Action (Revoke_Approved ---> Revoke_In_Progress)
+            try:
+                1) (in target namespace) Revoke access to the revoked tables to the redshift role in external schema (if schema exists)
+                2) (in target namespace) Revoke access to the revoked tables to the redshift role in self.local_db (if database exists)
+                3) (in source namespace) If that table is not shared in this namespace, remove table from datashare (if datashare exists)
+            except:
+                Update table status with Failure Action (Revoke_In_Progress ---> Revoke_Failed)
+        If the previous is successful, we proceed to clean-up shared resources across datashares:
+        try:
+            4) (in target namespace) If no more tables are shared with the redshift role, revoke usage access to the external schema to the redshift role
+            5) (in target namespace) If no more tables are shared with the redshift role, revoke usage access to the self.local_db to the redshift role
+            6) (in target namespace) If no more tables are shared with any role in this namespace, drop external schema
+            7) (in target namespace) If no more tables are shared with any role in this namespace, drop local database
+            8) (in source namespace) If no more tables are shared with any role in this namespace, drop datashare
+            # Drop datashare deletes it from source and target, alongside its permissions (for both same and cross account)
+            Update NON-FAILED tables with Success Action (Revoke_In_Progress ---> Revoke_Succeeded)
+        except:
+            Update tables with Failure Action (Revoke_In_Progress ---> Revoke_Failed)
+        Returns
+        -------
+        True if share is revoked successfully
+        """
+        log.info('##### Starting Cleaning up Redshift tables #######')
+        if not self.tables:
+            log.info('No Redshift tables to revoke. Skipping...')
+        else:
+            self._initialize_clients()
+            for table in self.tables:
+                log.info(f'Revoking access to table {table}...')
+                local_db_exists = self.redshift_data_client_in_target.check_database_exists(database=self.local_db)
+                # 1) (in target namespace) Revoke access to the revoked tables to the redshift role in external schema (if schema exists)
+                if local_db_exists and self.redshift_data_client_in_target.check_schema_exists(
+                    schema=self.external_schema, database=self.target_connection.database
+                ):
+                    execute_and_suppress_exception(
+                        func=self.redshift_data_client_in_target.revoke_select_table_access_to_redshift_role(
```

Comment:
you need to invoke it like `suppress(self.redshift_data_client_in_target, schema=self.extrernal_schema, table=table.name, ...)`, essentially defering the execution to the `suppress` util. Your current code will execute the method and then pass the returned value into the util thus if it throws it won't work.

Reply:
Uff, my bad, I was too fast and did not pay enough attention. It is fixed now but just in case I am doing some testing